### PR TITLE
Set up a "Hello world" project with rails-react gem

### PR DESCRIPTION
**Pull request by aliabbani**
In this milestone I:
- Created a new Rails app called 'hello-rails-react'.
- Initialized the project with Git.
- Set up your project with webpacker the react-rails gems.
- Made sure that your project has a Postgres database set up.
- Created a table for storing the messages and create 5 different greetings.
- Created an API endpoint that selects a random greeting from the table.
- Created a static view that will be the root of your app.
- Created the App component with react-router and render it in the static view.
- Created the Greeting component that will display a greeting. Set it up as a route in your App component.
- Created a store, an action and a reducer that will connect to you API endpont to get the random greeting.

### CODE REVIEW

File: app/javascript/components/App.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const store = configureStore();
+
+class App extends React.Component {
+  render () {
+    return (
+    <Provider store={store}>
+      <BrowserRouter>
+        <Routes>
+          <Route path='/' element={<Greeting title={'first greeting page path'} />} />
+          {/* <Route path='/greeting' element={<Greeting title={'greeting greeting page path'} />} /> */}
```
Comment (t-yanick) [inline app/javascript/components/App.js]: - [ ] Please kindly remove `commented code` from your `code base`, as it make your `code base` dirty and less readable.

File: app/javascript/components/Greeting.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import React from "react"
+// import PropTypes from "prop-types"
```
Comment (t-yanick) [inline app/javascript/components/Greeting.js]: - [ ] Please kindly remove `commented code` from your `code base`, as it make your `code base` dirty and less readable.

File: app/javascript/components/Greeting.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}
+const structuredSelector = createStructuredSelector({
+  greetings: (state) => state.greetings,
+});
+
+const mapDispatchToProps = { getGreetings };
+
+// Greeting.propTypes = {
+//   greeting: PropTypes.string,
+// };
```
Comment (t-yanick) [inline app/javascript/components/Greeting.js]: - [ ] Please kindly remove `commented code` from your `code base`, as it make your `code base` dirty and less readable.

File: app/javascript/redux/configureStore.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { applyMiddleware, createStore } from "redux";
+
+import thunk from "redux-thunk";
+// import { composeWithDevTools } from 'redux-devtools-extension';
```
Comment (t-yanick) [inline app/javascript/redux/configureStore.js]: - [ ] Please kindly remove `commented code` from your `code base`, as it make your `code base` dirty and less readable.

File: app/javascript/redux/configureStore.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+const initialState = {
+    greetings: []
+};
+
+function rootReducer(state, action) {
+    // console.log(action.type);
```
Comment (t-yanick) [inline app/javascript/redux/configureStore.js]: - [ ] Please kindly remove `commented code` from your `code base`, as it make your `code base` dirty and less readable.

**Review by t-yanick — APPROVED** [review]
Hi @aliabbani, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments below and under the review._

 ### I will leave the comment here, since your `Readme file` is not accessible for this PR ###

- [ ] Please kindly update your `readme file` to reflect the changes for this milestone as this will make your `readme file` more informative and professional. Checkout this [template](https://github.com/microverseinc/readme-template) and adapt it to your project.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi @aliabbani, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

- Let's update the readme file as advised by the previous reviewer before we can submit for re-review

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
## Hi @aliabbani, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by aliabbani** [comment]
> ## Hi @aliabbani,
> Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉
> 
> # Status: Approved ✔️
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**